Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md

```diff
@@ -0,0 +1,105 @@
+# Bit Manipulation
+
+## Use of Bit Operators
+
+Bit operators perform operations much faster than arthmetic or logical operators
+
+## Types of Bit Operator
+
+There are mainly 6 types of bit operators
+
+1. `AND`
+2. `OR`
+3. `XOR`
+4. `Left shift operator`
+5. `Right shift operator`
+6. `Bitwise NOT`
+
+   <br/><br>
+
+   ![Symbols representing Bitwise operator](https://miro.medium.com/max/540/1*lwKj-TpvToBxuq98LXII1A.png)
+
+<br/><br>
+
+## **_Bitwise AND_**
+
+The bitwise AND operator takes two arguments and performs AND operation on the operands bit by bit.
+The operation will only be 1 when both operands are 1 otherwise it remains 0.
+
+![Table](https://i1.wp.com/techvidvan.com/tutorials/wp-content/uploads/sites/2/2019/12/implementation-of-and-operation-on-binary-digits.jpg?resize=341%2C272&ssl=1)
+
+<br />
+
+## **_Bitwise OR_**
+
+The bitwise OR operator takes two arguments and performs OR operation on each bit of the value of the operand. In the OR operation in binary, the resultant is 0 when both operands are 0 otherwise it is 1.
+
+![](https://i2.wp.com/techvidvan.com/tutorials/wp-content/uploads/sites/2/2019/12/bitwise-or-operations.jpg?resize=325%2C238&ssl=1)
+
+<br />
+
+## **_Bitwise XOR_**
+
+The bitwise Xor takes two operands and performs Xor operation on the binary digits. The Xor operation gives 0 as a result when both operands are the same and 1 when operands are different.
+
+![](https://i2.wp.com/techvidvan.com/tutorials/wp-content/uploads/sites/2/2019/12/example-of-bitwise-Xor-operation.jpg?resize=500%2C197&ssl=1)
+
+<br/>
+
+## **_Left Shift Operator_**
+
+The left shift operator shifts the bits of the numbers by a specified number of places.
+
+`Example 1`:
+
+Let’s see how this works.
+
+**2<<2 = 8**
+
+2 is represented as ‘010’
+
+We shift the digits to _left up_ to two places. So, it will be ‘1000’ in binary and 8 in decimal.
+
+`Example 2`:
+
+**3<<4 = 48**
+
+10 in binary is ‘1010’
+
+Shifting 4 digits to the _left_ will make it ‘011 0000’ which is 48 in decimal.
+
+Shortcut for \*\*left shit operator
+
+<br />
+
+## **_Right Shift Operator_**
+
+This is similar to the left shift, but instead of shifting digits to the left we are shifting them to the right. In this process, some of the bits are lost.
```

Comment:
```suggestion
This is similar to the left shift, but instead of shifting digits to the left, we shift them to the right. In this process, some of the bits get discarded.
```